dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/codec/ExchangeCodec.java

```diff
@@ -175,25 +172,35 @@ protected Object decodeBody(Channel channel, InputStream is, byte[] header) thro
             Request req = new Request(id);
             req.setVersion(Version.getProtocolVersion());
             req.setTwoWay((flag & FLAG_TWOWAY) != 0);
-            if ((flag & FLAG_EVENT) != 0) {
-                req.setEvent(Request.HEARTBEAT_EVENT);
-            }
+            Object data = null;
             try {
                 ObjectInput in = CodecSupport.deserialize(channel.getUrl(), is, proto);
-                Object data;
                 if (req.isHeartbeat()) {
                     data = decodeHeartbeatData(channel, in);
                 } else if (req.isEvent()) {
                     data = decodeEventData(channel, in);
                 } else {
                     data = decodeRequestData(channel, in);
                 }
-                req.setData(data);
             } catch (Throwable t) {
                 // bad request
                 req.setBroken(true);
                 req.setData(t);
             }
+            if (!req.isBroken()) {
+                if ((flag & FLAG_EVENT) != 0) {

```

Comment:
In case if the condition `if ((flag & FLAG_EVENT) != 0)` does not gets satisfied then, what will be the event value of request object? Is there any default event type?

Reply:
With the limited knowledege about this part (so @chickenlj  @carryxyh @beiwei30  correct me if I am not correct). I think we should either have a default event type if noting is passed in data(more informative way and declaring a default behaviour) or we should reject the request if no default behaviour is spported.  It will make the easy to debug and reason about the issue. 

If we provide some default behaviour (event type) and which is not communicated, for user it might be confusiong (because it is not documented or communicated to them).